pkg/audit/stats_reporter.go

```diff
@@ -0,0 +1,124 @@
+package audit
+
+import (
+	"context"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/metrics"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	totalViolationsName  = "violations_total"
+	constraintsTotalName = "constraints_total"
+	auditLatency         = "audit_latency"
+	methodType           = "audit"
+)
+
+var (
+	violationsTotalM  = stats.Int64(totalViolationsName, "Total number of violations per constraint", stats.UnitNone)
+	constraintsTotalM = stats.Int64(constraintsTotalName, "Total number of enforced constraints", stats.UnitNone)
+	auditLatencyM     = stats.Float64(auditLatency, "Latency of audit operation", stats.UnitMilliseconds)
+
+	methodTypeKey     = tag.MustNewKey("method_type")
+	constraintKindKey = tag.MustNewKey("constraint_kind")
+	constraintNameKey = tag.MustNewKey("constraint_name")
+)
+
+func init() {
+	register()
+}
+
+func register() {
```

Comment:
What do you think about also exporting a metric for the timestamp of the last audit run? It could be useful for an alerting rule that makes sure that more the N minutes don't pass between any two runs.

Reply:
this is documented in #157 to be one of the metrics to be implemented

Reply:
Great. I wrote up a separate design for the metrics a little while ago so I just copied my notes there for completeness.